src/java/picard/util/IntervalListScatterer.java

```diff
@@ -63,6 +74,20 @@ public int compare(final Interval o1, final Interval o2) {
     public List<IntervalList> scatter(final IntervalList uniquedIntervalList, final int scatterCount) {
         return scatter(uniquedIntervalList, scatterCount, false);
     }
+    
+    /** Helper for the scatter method */
+    private boolean shouldAddToRunningIntervalList(final long idealSplitLength, final long projectedSize, final double projectedSizeOfRemainingDivisions) {
+        switch (mode) {
+            case INTERVAL_SUBDIVISION:
+            case BALANCING_WITHOUT_INTERVAL_SUBDIVISION:
+                return (projectedSize <= idealSplitLength);
+            case BALANCING_WITHOUT_INTERVAL_SUBDIVISION_WITH_OVERFLOW:
+                return (projectedSize <= idealSplitLength || idealSplitLength < projectedSizeOfRemainingDivisions);
+            default:
+                throw new PicardException("Unknown mode: " + mode.name());
```

Comment:
This default really would never happen. Might be better to structure it with 
      case BALANCING_WITHOUT_INTERVAL_SUBDIVISION_WITH_OVERFLOW:
                return (projectedSize <= idealSplitLength || idealSplitLength < projectedSizeOfRemainingDivisions);
      default:
                return (projectedSize <= idealSplitLength);